### PR TITLE
feat: Introduce workspace `rust-analyzer.toml`s

### PR DESCRIPTION
**Pull request by alibektas**
In order to globally configure a project it was, prior to this PR, possible to have a `ratoml` at the root path of a project. This is not the case anymore. Instead we now let ratoml files that are placed at the root of any workspace have a new scope called `workspace`. Although there is not a difference between a `workspace` scope and and a `global` scope, future PRs will change that.

### CODE REVIEW

File: crates/rust-analyzer/src/config.rs [hunk 1]
```diff
@@ -1333,11 +1354,6 @@ impl Config {
         // FIXME @alibektas : Temporary solution. I don't think this is right as at some point we may allow users to specify
         // custom USER_CONFIG_PATHs which may also be relative.
         let user_config_path = VfsPath::from(AbsPathBuf::assert(user_config_path));
```
Comment (Veykril) [inline crates/rust-analyzer/src/config.rs:1356]: On a follow up note, I feel like we should not make this configurable, where would you even configure this? I guess in the lient config but that still feels a bit odd?

Comment (alibektas) [reply crates/rust-analyzer/src/config.rs:1356]: by setting a command line argument to the server but ok I will remove it

Comment (alibektas) [reply crates/rust-analyzer/src/config.rs:1358]: I took a note of it and will make a separate PR for this.

**Review by Veykril — APPROVED** [review]

**Pull request merged**